File: nodejs-advanced/src/utils/validator.js
function isValidEmail(email) {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
}

function isValidName(name) {
  return name && name.length >= 2 && name.length <= 100;
}

module.exports = {
  isValidEmail,
  isValidName
};

File: java-advanced/src/main/java/functions/utils/Validator.java
package functions.utils;

import java.util.regex.Pattern;

public class Validator {
    private static final Pattern EMAIL_PATTERN =
        Pattern.compile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$");

    public boolean isValidEmail(String email) {
        return email != null && EMAIL_PATTERN.matcher(email).matches();
    }

    public boolean isValidName(String name) {
        return name != null && name.length() >= 2 && name.length() <= 100;
    }
}
